Cannot create a batch of xtl::xcomplex

Using current HEAD or latest tags (8.1.0 for xsimd and 0.7.4 for xtl), creating a `xsimd::batch<xtl::xcomplex<double>>` yields a compile-time error.

Minimal working example:
```c++
#define XSIMD_ENABLE_XTL_COMPLEX 1
#include <xsimd/xsimd.hpp>

int main()
{
    xsimd::batch<xtl::xcomplex<double>> foo;
    return 0;
}
```

The compile error is, using `g++ -I xtl/include -I xsimd/include -std=c++14 -c foo.cpp` is:
```
In file included from xsimd/include/xsimd/memory/../config/../types/../types/./././././xsimd_sse2_register.hpp:16:0,
                 from xsimd/include/xsimd/memory/../config/../types/../types/././././xsimd_sse3_register.hpp:15,
                 from xsimd/include/xsimd/memory/../config/../types/../types/./././xsimd_ssse3_register.hpp:15,
                 from xsimd/include/xsimd/memory/../config/../types/../types/././xsimd_sse4_1_register.hpp:15,
                 from xsimd/include/xsimd/memory/../config/../types/../types/./xsimd_sse4_2_register.hpp:15,
                 from xsimd/include/xsimd/memory/../config/../types/../types/xsimd_fma3_sse_register.hpp:15,
                 from xsimd/include/xsimd/memory/../config/../types/xsimd_all_registers.hpp:12,
                 from xsimd/include/xsimd/memory/../config/xsimd_arch.hpp:19,
                 from xsimd/include/xsimd/memory/xsimd_aligned_allocator.hpp:27,
                 from xsimd/include/xsimd/xsimd.hpp:41,
                 from foo.cpp:2:
xsimd/include/xsimd/memory/../config/../types/../types/././././././xsimd_register.hpp: In instantiation of ‘struct xsimd::types::simd_register<xtl::xcomplex<double>, xsimd::sse2>’:
xsimd/include/xsimd/types/xsimd_batch.hpp:46:11:   required from ‘class xsimd::batch<xtl::xcomplex<double> >’
foo.cpp:6:41:   required from here
xsimd/include/xsimd/memory/../config/../types/../types/././././././xsimd_register.hpp:31:13: error: static assertion failed: usage of simd_register with unsupported type
             static_assert(!Arch::supported() || has_simd_register<T, Arch>::value,
             ^~~~~~~~~~~~~
In file included from xsimd/include/xsimd/xsimd.hpp:46:0,
                 from foo.cpp:2:
xsimd/include/xsimd/types/xsimd_batch.hpp: In instantiation of ‘class xsimd::batch<xtl::xcomplex<double> >’:
foo.cpp:6:41:   required from here
xsimd/include/xsimd/types/xsimd_batch.hpp:53:81: error: no type named ‘register_type’ in ‘struct xsimd::types::simd_register<xtl::xcomplex<double>, xsimd::sse2>’
         using register_type = typename types::simd_register<T, A>::register_type; ///< SIMD register type abstracted by this batch.
```

With xsimd 7.6.0, a similar example (replacing the batch declaration with `xsimd::simd_type<xtl::xcomplex<double>> foo` works.

Note: I've gotten there when compiling xtensor with `-DXTENSOR_USE_XSIMD=ON` in the `test_xcomplex` target. Let me know if this issue should be reported to xtensor.

xtensor still fails with #772 , showing the error that got set by the patch:
```
[ 38%] Building CXX object test/CMakeFiles/test_xdatesupport.dir/test_xdatesupport.cpp.o
cd /home/drew/projects/mathlibs/build/xtensor/obj-x86_64-linux-gnu/test && /usr/bin/c++ -DXSIMD_ENABLE_XTL_COMPLEX -DXTENSOR_USE_XSIMD -I/home/drew/projects/mathlibs/build/xtensor/include -g -O2 -ffile-prefix-map=/home/drew/projects/mathlibs/build/xtensor=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -DXSIMD_ENABLE_XTL_COMPLEX=1 -march=native -std=c++14 -Wunused-parameter -Wextra -Wreorder -Wconversion -Wno-sign-conversion  -Wold-style-cast -Wunused-variable -ftemplate-backtrace-limit=0 -MD -MT test/CMakeFiles/test_xdatesupport.dir/test_xdatesupport.cpp.o -MF CMakeFiles/test_xdatesupport.dir/test_xdatesupport.cpp.o.d -o CMakeFiles/test_xdatesupport.dir/test_xdatesupport.cpp.o -c /home/drew/projects/mathlibs/build/xtensor/test/test_xdatesupport.cpp
In file included from /usr/include/xsimd/xsimd.hpp:29,
                 from /home/drew/projects/mathlibs/build/xtensor/include/xtensor/xtensor_config.hpp:61,
                 from /home/drew/projects/mathlibs/build/xtensor/test/test_common_macros.hpp:9,
                 from /home/drew/projects/mathlibs/build/xtensor/test/test_xcomplex.cpp:10:
/usr/include/xsimd/types/xsimd_batch.hpp: In instantiation of ‘class xsimd::batch<xtl::xcomplex<double>, xsimd::fma3<xsimd::avx2> >’:
/usr/include/xsimd/arch/generic/xsimd_generic_details.hpp:167:66:   required by substitution of ‘template<class A, class From, class To> struct xsimd::kernel::detail::conversion_type_impl<A, From, To, typename xsimd::detail::make_void<decltype (xsimd::kernel::detail::fast_cast(declval<const xsimd::batch<T, A>&>(), declval<const xsimd::batch<Tp, A>&>(), declval<const A&>()))>::type> [with A = xsimd::fma3<xsimd::avx2>; From = xtl::xcomplex<double>; To = std::complex<double>]’
/usr/include/xsimd/arch/generic/xsimd_generic_details.hpp:175:19:   required by substitution of ‘template<class A, class From, class To> using conversion_type = typename xsimd::kernel::detail::conversion_type_impl<A, From, To>::type [with A = xsimd::fma3<xsimd::avx2>; From = xtl::xcomplex<double>; To = std::complex<double>]’
/usr/include/xsimd/arch/generic/xsimd_generic_memory.hpp:151:43:   required from ‘xsimd::batch<Tp, A> xsimd::kernel::load_aligned(const T_in*, xsimd::kernel::convert<T_out>, xsimd::kernel::requires_arch<xsimd::generic>) [with A = xsimd::fma3<xsimd::avx2>; T_in = xtl::xcomplex<double>; T_out = std::complex<double>; xsimd::kernel::requires_arch<xsimd::generic> = const xsimd::generic&]’
/usr/include/xsimd/types/xsimd_api.hpp:1020:39:   required from ‘xsimd::simd_return_type<From, To> xsimd::load_as(const From*, xsimd::aligned_mode) [with To = xtl::xcomplex<double>; A = xsimd::fma3<xsimd::avx2>; From = xtl::xcomplex<double>; xsimd::simd_return_type<From, To> = xsimd::batch<std::complex<double>, xsimd::fma3<xsimd::avx2> >]’
/home/drew/projects/mathlibs/build/xtensor/include/xtensor/xcontainer.hpp:751:48:   required from ‘xt::container_simd_return_type_t<typename xt::xcontainer_inner_types<D>::storage_type, typename xt::xcontainer_inner_types<D>::storage_type::value_type, requested_type> xt::xcontainer<D>::load_simd(xt::xcontainer<D>::size_type) const [with align = xsimd::aligned_mode; requested_type = xtl::xcomplex<double>; long unsigned int N = 4; D = xt::xarray_container<xt::uvector<xtl::xcomplex<double>, xsimd::aligned_allocator<xtl::xcomplex<double>, 32> >, xt::layout_type::row_major, xt::svector<long unsigned int, 4, std::allocator<long unsigned int>, true>, xt::xtensor_expression_tag>; xt::container_simd_return_type_t<typename xt::xcontainer_inner_types<D>::storage_type, typename xt::xcontainer_inner_types<D>::storage_type::value_type, requested_type> = xsimd::batch<std::complex<double>, xsimd::fma3<xsimd::avx2> >; typename xt::xcontainer_inner_types<D>::storage_type::value_type = xtl::xcomplex<double>; typename xt::xcontainer_inner_types<D>::storage_type = xt::uvector<xtl::xcomplex<double>, xsimd::aligned_allocator<xtl::xcomplex<double>, 32> >; xt::xcontainer<D>::size_type = long unsigned int]’
/home/drew/projects/mathlibs/build/xtensor/test/test_xcomplex.cpp:327:127:   required from here
/usr/include/xsimd/types/xsimd_batch.hpp:405:46: error: static assertion failed: Please use batch<std::complex<T>, A> initialized from xtl::complex instead
  405 |         static_assert(std::is_same<T, void>::value,
      |                                              ^~~~~
``` 

I guess it means xtensor still needs to be patched to `std::complex` as indicated by the message?

--- a/include/xsimd/types/xsimd_batch.hpp
+++ b/include/xsimd/types/xsimd_batch.hpp
@@ -425,6 +425,14 @@ namespace xsimd
     template <class T, class A>
     constexpr std::size_t batch<std::complex<T>, A>::size;
 
+#ifdef XSIMD_ENABLE_XTL_COMPLEX
+    template <typename T, bool i3ec, typename A>
+    struct batch<xtl::xcomplex<T, T, i3ec>, A>
+    {
+        static_assert(std::is_same<T, void>::value,
+                      "Please use batch<std::complex<T>, A> initialized from xtl::complex instead");
+    };
+#endif
 }
 
 #include "../arch/xsimd_isa.hpp"

